### PR TITLE
Update dataset docstrings

### PR DESCRIPTION
**Pull request by nd-02110114**
## What I did

- Updated docstrings in dataset classes (80% of diff)
- Replaced `get_image` function to `ImageLoader. _get_image`.
  - This is because  `get_image` is defined in a many places of ImageLoader.
- Removed flake8 error in dataset test and support.py files.

### CODE REVIEW

File: deepchem/data/datasets.py [hunk 1]
```diff
@@ -703,17 +723,11 @@ def __init__(self,
...
-    Get the number of elements in the dataset.
-    """
+    """Get the number of elements in the dataset."""
     return len(self._y)
 
   def get_shape(self) -> Tuple[Shape, Shape, Shape, Shape]:
-    """Get the shape of the dataset.
-
-    Returns four tuples, giving the shape of the X, y, w, and ids
-    arrays.
```
Comment (peastman) [inline deepchem/data/datasets.py:715]: Why remove this?

File: deepchem/data/datasets.py [hunk 2]
```diff
@@ -346,44 +365,35 @@ def iterbatches(self,
...
   def itersamples(self) -> Iterator[Batch]:
-    """Get an object that iterates over the samples in the dataset.
-
-    Example:
-
-    >>> dataset = NumpyDataset(np.ones((2,2)))
-    >>> for x, y, w, id in dataset.itersamples():
-    ...   print(x.tolist(), y.tolist(), w.tolist(), id)
-    [1.0, 1.0] [0.0] [0.0] 0
-    [1.0, 1.0] [0.0] [0.0] 1
```
Comment (peastman) [inline deepchem/data/datasets.py:378]: Why remove the example?

File: deepchem/data/datasets.py [hunk 3]
```diff
@@ -2109,16 +2274,16 @@ def select(self, indices: Sequence[int],
...
-    select_dir: Optional[str], (default None)
+    select_dir: str, optional (default None)
       Path to new directory that the selected indices will be copied
       to.
 
     Returns
     -------
     DiskDataset
-      Contains selected indices.
+      A selected DiskDataset object
```
Comment (peastman) [inline deepchem/data/datasets.py:2286]: How about, "A Dataset containing the selected samples"?

File: deepchem/data/datasets.py [hunk 4]
```diff
@@ -2465,29 +2638,43 @@ def iterate(dataset, batch_size, epochs, deterministic, pad_batches):
...
-  def transform(self, transformer: "dc.trans.Transformer",
-                **args) -> NumpyDataset:
+    return ((self._get_image(self._X, i), self._get_image(self._y, i),
+             self._w[i], self._ids[i]) for i in range(n_samples))
+
+  def transform(
+      self,
+      transformer: "dc.trans.Transformer",
+      **args,
+  ) -> "ImageDataset":
```
Comment (peastman) [inline deepchem/data/datasets.py:2677]: The transformed dataset is a NumpyDataset, not an ImageDataset.

File: deepchem/data/datasets.py [hunk 5]
```diff
@@ -2500,28 +2687,34 @@ def transform(self, transformer: "dc.trans.Transformer",
...
     Returns
     -------
-    a newly constructed Dataset object
+    ImageDataset
+      A newly constructed ImageDataset object
     """
     newx, newy, neww, newids = transformer.transform_array(
         self.X, self.y, self.w, self.ids)
-    return NumpyDataset(newx, newy, neww, newids)
+    return ImageDataset(newx, newy, neww, newids)
```
Comment (peastman) [inline deepchem/data/datasets.py:2700]: This is incorrect.  `transform_array()` returns numpy arrays.

Comment (nd-02110114) [reply deepchem/data/datasets.py]: OK. I revert this modifications

File: deepchem/data/datasets.py (hunk 4 above)
Comment (nd-02110114) [reply deepchem/data/datasets.py]: OK. I revert this modification

File: deepchem/data/datasets.py (hunk 3 above)
Comment (nd-02110114) [reply deepchem/data/datasets.py]: I fixed using it!

File: deepchem/data/datasets.py (hunk 2 above)
Comment (nd-02110114) [reply deepchem/data/datasets.py]: I revert this modification

File: deepchem/data/datasets.py (hunk 1 above)
Comment (nd-02110114) [reply deepchem/data/datasets.py:721]: I revert this modification

File: deepchem/models/torch_models/cgcnn.py [hunk 6]
```diff
@@ -26,11 +26,11 @@ class CGCNNLayer(nn.Module):
   41
   >>> cgcnn_dgl_graph = cgcnn_graph.to_dgl_graph()
   >>> print(type(cgcnn_dgl_graph))
-  <class 'dgl.graph.DGLGraph'>
+  <class 'dgl.heterograph.DGLHeteroGraph'>
```
Comment (nd-02110114) [inline deepchem/models/torch_models/cgcnn.py:29]: The fix is related to the new version release. (DGL. 0.5.0 released today). I seem the master branch CI also failed.

**Comment by nd-02110114** [comment]
I think this PR is a ready to a second review. This PR also fixed the CGCNN doctest error.

**Comment by peastman** [comment]
Looks good!

**Pull request merged**